Enhance pbench server unit tests

portante opened the pull request:
A series of commits to enhance the server unit tests now that server activation and auditing are used.

portante commented:
Depends on #939.

portante commented:
FYI, `test-5.2` fails intentionally.  The gold file represents the behavior it seems we should have, but don't just yet.

portante commented:
The unit tests no longer fail intentionally, all behaviors have been fixed and any unit test failure should indicate a problem.

portante commented:
@tenstormavi, can you make it a priority to review this PR now?

tenstormavi commented on server/pbench/bin/pbench-sync-satellite:
```diff
@@ -211,43 +246,28 @@ for host in $hosts ;do
         state_list=$(cat $logdir/$host/ok-checks.log | sed -n 's/: OK//p')
         for x in $state_list; do
-            echo "$host/TO-SYNC/$x" >> $LOGSDIR/change_state.$prefix.log
+            echo "$host/TO-SYNC/$x" >> $LOGSDIR/change_state.$remote_prefix.log
```
According to the new naming convention, this line should be

`echo "$host/TO-SYNC/$x" >> ${state_change_log}` 

otherwise, it won't be saving the results in the right position. And we won't get anything while checking whether to call the change-state script or not.

tenstormavi replied:
According to the new naming convention, this line should be

`echo "$host/TO-SYNC/$x" >> ${state_change_log}` 

otherwise, it won't be saving the results in the right position. And we won't get anything while checking whether to call the change-state script or not.

tenstormavi replied:
According to the new naming convention, this line should be

`echo "$host/TO-SYNC/$x" >> ${state_change_log}` 

otherwise, it won't be saving the results in the right position. And we won't get anything while checking whether to call the change-state script or not.

tenstormavi commented on server/pbench/bin/pbench-sync-satellite:
```diff
@@ -211,43 +246,28 @@ for host in $hosts ;do
         state_list=$(cat $logdir/$host/ok-checks.log | sed -n 's/: OK//p')
         for x in $state_list; do
-            echo "$host/TO-SYNC/$x" >> $LOGSDIR/change_state.$prefix.log
+            echo "$host/TO-SYNC/$x" >> $LOGSDIR/change_state.$remote_prefix.log
```
According to the new naming convention, this line should be

`echo "$host/TO-SYNC/$x" >> ${state_change_log}` 

otherwise, it won't be saving the results in the right position. And we won't get anything while checking whether to call the change-state script or not.

portante replied:
Thanks, I'll fix this in an update.

tenstormavi commented:
@portante PTAL 

And sorry GitHub was down and I tried multiple times to review the PR but that time it was showing error and suddenly then it bombarded this PR with all my reviews. Tried to clean the mess but not succeeded completely.

portante commented:
@tenstormavi, it seems that things are cleaning themselves up now, thanks for the review, and good catch!

portante commented:
Now depends on #965 being merged first.

portante commented:
A few updates pushed:

 * Fixed `pbench-gen-json-payload` to remove white space at the end of lines (so we don't get warnings in our patches during a rebase)
 * Renamed `pbench-audit-archive` to `pbench-audit-server` so that we can have it audit more than just the archive tree

Both of these changes affected most, if not all, the gold files, and since this big patch touches all the gold files I figured it was best to make these kinds of changes here.

portante commented on server/pbench/lib/config/pbench-server.cfg.example:
```diff
@@ -5,45 +5,47 @@
-dispatch-states = TO-UNPACK, TO-INDEX
+# Backup dir may be remote
+backup-host = pbench-backup.example.com
+pbench-backup-dir = %(default-user)s@%(deploy-backup-host)s:/srv/pbench.archive.backup
```
Change `deploy-backup-host` to `backup-host`.

portante replied:
Change `default-user` to `user`.

portante replied:
Applied both suggestions above.  Should be in forth-coming patch.

portante commented on server/pbench/lib/config/pbench-server.cfg.example:
```diff
@@ -5,45 +5,47 @@
+# remote satellite pbench servers.
 [pbench-results]
 host = %(default-host)s
 mailfrom = %(default-user)s@%(host)s
```
Change `default-user` to `user`.

portante commented on server/pbench/lib/config/pbench-server.cfg.example:
```diff
@@ -5,45 +5,47 @@
+# In this example we extend the pbench-results role to include sync'ing with
+# remote satellite pbench servers.
 [pbench-results]
 host = %(default-host)s
```
Remove `host = ...`.

portante commented on server/pbench/lib/config/pbench-server.cfg.example:
```diff
@@ -5,45 +5,47 @@
 [config]
-paths = /opt/pbench-server/lib/config
+path = /opt/pbench-server/lib/config
 files = pbench-server-default.cfg.example
```
Remove `.example` when you rename the default config file.

portante commented on server/pbench/bin/state/config/pbench-server.cfg:
```diff
@@ -1,151 +1,53 @@
 ###########################################################################
 ## runtime section
 ###########################################################################
 [sosreport]
```
We can remove this section now ...

portante replied:
Removed, should be available in the next patch.

portante commented on server/pbench/bin/state/config/pbench-server.cfg:
```diff
@@ -1,151 +1,53 @@
-[pbench-backup]
+# The definition of the crontab role for sync'ing satellite pbench servers.
+[pbench-sync-satellites]
 host = %(default-host)s
```
Remove `host = ...`

portante replied:
Removed, should be available in the next patch.

portante replied:
Well, no, we actually need this here like the other default roles that have it.  No change being made.

portante commented on server/pbench/bin/state/config/pbench-server.cfg:
```diff
@@ -1,151 +1,53 @@
-satellite-archive = /path/to/pbench/archive/fs-version-001
+# Template values for this satellite
+[satellite-one]
+satellite-host = pbench-satellite.example.com
```
Add comment that this name is what is expected in the `ssh` mock command.

portante replied:
Added a comment, should be available in the next patch.

portante commented on server/pbench/bin/pbench-unpack-tarballs:
```diff
@@ -64,11 +64,9 @@ test -d $ARCHIVE || doexit "Bad ARCHIVE=$ARCHIVE"
-    test -d $UNPACK_PATH || doexit "Bad UNPACK_PATH=$UNPACK_PATH"
-fi
+
+UNPACK_PATH=$(getconf.py pbench-unpack-dir pbench-server)
```
Remove argument to `pbench-unpack-tarballs`.

portante replied:
Fixed in upcoming patch.

portante commented on server/pbench/bin/pbench-remote-satellite-state-change:
```diff
@@ -0,0 +1,48 @@
+        :
+        ;;
+    *)
+        echo "Usage: $PROG <satellite-config>"
```
Missing second argument.

portante replied:
Fixed in upcoming patch.

portante commented on server/pbench/bin/pbench-remote-sync-package-tarballs:
```diff
@@ -0,0 +1,43 @@
+        :
+        ;;
+    *)
+        echo "Usage: $PROG <satellite-config>"
```
Missing second argument...

portante replied:
Fixed in upcoming patch.

ndokos commented:
FWIW, unit tests passed on F27.

portante commented on server/pbench/bin/unittests:
```diff
@@ -114,6 +115,10 @@ function _run_allscripts {
     _run pbench-verify-backup-tarballs
+    # Invoke pbench-satellite-cleanup in the context of the satellite
+    _old_CONFIG=${CONFIG}
+    CONFIG=${SATCONFIG} _run pbench-satellite-cleanup
```
Fixed this to be explicit; updated in a forth-coming patch.

portante commented on server/pbench/bin/unittests:
```diff
@@ -91,21 +98,15 @@ function _run_activate {
     _run pbench-server-prep-shim-001
     _run pbench-server-prep-shim-002
-    _run pbench-sync-satellite UT remote.example.com /pbench/archive/fs-version-001
+    _run pbench-sync-satellite satellite-one
```
Done; should be there in forth-coming patch.

portante commented on server/pbench/bin/pbench-server-activate-web-server:
```diff
@@ -1,8 +1,7 @@
 #! /bin/bash
 
-pbench_install_dir=$(getconf.py install-dir pbench-server)
```
Removed `pbench-server-activate-web-server` entirely, and added the setup of the static directory symlink in the `pbench-server-activate-start-httpd` script.

The forth coming patch should contain the code to address all this.

portante commented:
@ndokos, I'll wait for your official review before merging.

ndokos commented on server/pbench/lib/config/pbench-server-satellite.cfg.example:
```diff
@@ -0,0 +1,29 @@
+# This file describes the elements of deploying the pbench background tools.
+[DEFAULT]
+# cvalues here override those in pbench-server-default.cfg.
```
Is this a typo? What does "cvalues" mean?

portante replied:
Changed to "The values ...".

ndokos commented on server/pbench/bin/state/config-satellite/pbench-server.cfg:
```diff
@@ -0,0 +1,38 @@
+# A satellite just needs to prep and process results, no backup necessary
+roles = pbench-prep, pbench-results
+use-mail-for-status = yes
+realhost = pbench.example.com
```
Do we need an explanation here? Why is `realhost` required?

portante replied:
Added a comment explaining why it is needed.
